packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -25,22 +32,126 @@ impl PostgresDatabase {
     }
 }
 
+impl PostgresDatabase {
+    async fn insert_version_id(&self, id: &Identifier) -> Result<(), DatastoreError> {
+        sqlx::query(r#"INSERT INTO ids (version_id, base_id) VALUES ($1, $2);"#)
+            .bind(&id.version_id)
+            .bind(&id.base_id)
+            .execute(&self.pool)
+            .await
+            .report()
+            .change_context(DatastoreError)
+            .attach_printable_lazy(|| id.clone())
+            .attach_printable("Could not insert id")?;
+
+        Ok(())
+    }
+
+    async fn insert_data_type(
+        &self,
+        id: &Identifier,
+        data_type: &DataType,
+        created_by: AccountId,
+    ) -> Result<DataTypeRow, DatastoreError> {
+        sqlx::query_as(
+            r#"
+            INSERT INTO data_types (
+                version_id,
+                schema,
+                created_by
+            ) 
+            VALUES ($1, $2, $3)
+            RETURNING version_id, schema, created_by;
+            "#,
+        )
+        .bind(&id.version_id)
+        .bind(&data_type)
+        .bind(&created_by)
+        .fetch_one(&self.pool)
+        .await
+        .report()
+        .change_context(DatastoreError)
+        .attach_printable("Could not insert data type")
+    }
+}
+
 #[async_trait]
 impl Datastore for PostgresDatabase {
-    async fn create_data_type() -> Result<(), DatastoreError> {
-        todo!()
+    async fn create_data_type(
+        &self,
+        data_type: DataType,
+        created_by: AccountId,
+    ) -> Result<Qualified<DataType>, DatastoreError> {
+        let id = Identifier {
```

Comment:
Happy to leave this as is for this PR because it'll be easier to swap in another implementation, but one of the benefits of moving this into a method (outside of this) is that we can reuse it for the other "create" methods, and importantly we want to think about sortable version_ids if we can. As such, the `new` method would need to know about the older one (if it's not time based) so it can increment.

Lots to unpack, I don't think we should do anything in this PR about it.